packages/filesystem/src/electron-browser/file-dialog/electron-file-dialog-service.ts

```diff
@@ -91,13 +91,17 @@ export class ElectronFileDialogService extends DefaultFileDialogService {
     }
 
     protected async canRead(uris: MaybeArray<URI>): Promise<boolean> {
-        const inaccessibleFilePaths = await Promise.all((Array.isArray(uris) ? uris : [uris]).map(
-            async uri => (!await this.fileService.access(uri, FileAccess.Constants.R_OK) && uri.path || '')
-        ).filter(e => e));
-        if (inaccessibleFilePaths.length) {
-            this.messageService.error(`Cannot read ${inaccessibleFilePaths.length} resources: ${inaccessibleFilePaths.join(', ')}`);
+        const resources = Array.isArray(uris) ? uris : [uris];
+        const unreadableResources: URI[] = [];
+        await Promise.all(resources.map(async resource => {
+            if (!await this.fileService.access(resource, FileAccess.Constants.R_OK)) {
+                unreadableResources.push(resource);
```

Comment:
Do we definitely want to use the full URI rather than the path?

```suggestion
        const unreadableResources: string[] = [];
        await Promise.all(resources.map(async resource => {
            if (!await this.fileService.access(resource, FileAccess.Constants.R_OK)) {
                unreadableResources.push(resource.path.toString());
```

Reply:
@colin-grant-work I've updated the pull-request to store the `path` directly and not the entire `uri` as suggested :+1: